### PR TITLE
branch 'Block' tests integration

### PR DESCRIPTION
**Pull request by winsvega**
Branch "Block"  stateTests integration

### CODE REVIEW

**Comment by CJentzsch** [comment]
the buildbot says it doesn't compile.

**Comment by winsvega** [comment]
still in development

File: libethereum/Account.cpp [hunk 1]
```diff
@@ -48,17 +48,15 @@ AccountMap dev::eth::jsonToAccountMap(std::string const& _json, AccountMaskMap*
...
-		u256 balance;
+		u256 balance = 0;
 
-		bool haveBalance = true;
+		bool haveBalance = (o.count("wei") || o.count("finney") || o.count("balance"));
 		if (o.count("wei"))
 			balance = u256Safe(o["wei"].get_str());
 		else if (o.count("finney"))
 			balance = u256Safe(o["finney"].get_str()) * finney;
 		else if (o.count("balance"))
```
Comment (CJentzsch) [inline libethereum/Account.cpp]: you can do a `if (o.count("wei") ||  o.count("balance")) above, and remove that part.

File: libethereum/Account.cpp [hunk 2]
```diff
@@ -70,10 +68,12 @@ AccountMap dev::eth::jsonToAccountMap(std::string const& _json, AccountMaskMap*
 			ret[a] = Account(balance, Account::NormalCreation);
 
 		bool haveStorage = o.count("storage");
+		if (haveStorage)
 		for (pair<string, js::mValue> const& j: o["storage"].get_obj())
```
Comment (CJentzsch) [inline libethereum/Account.cpp]: indenting looks wrong

File: libethereum/Account.cpp [hunk 3]
```diff
@@ -70,10 +68,12 @@ AccountMap dev::eth::jsonToAccountMap(std::string const& _json, AccountMaskMap*
...
 			ret[a].setStorage(u256(j.first), u256(j.second.get_str()));
 
 		bool haveNonce = o.count("nonce");
+		if (haveNonce)
 		for (auto i = 0; i < u256Safe(o["nonce"].get_str()); ++i)
```
Comment (CJentzsch) [inline libethereum/Account.cpp]: same

File: libethereum/Account.h [hunk 4]
```diff
@@ -228,7 +228,6 @@ class AccountMask
 
 	bool allSet() const { return m_hasBalance && m_hasNonce && m_hasCode && m_hasStorage; }
 
-private:
```
Comment (CJentzsch) [inline libethereum/Account.h]: instead of removing `private`, define getter/setter in case you need direct access

File: test/TestHelper.cpp [hunk 5]
```diff
@@ -148,13 +165,17 @@ void ImportTest::importEnv(json_spirit::mObject& _o)
 	m_envInfo.setNumber(toInt(_o["currentNumber"]));
 	m_envInfo.setGasLimit(toInt(_o["currentGasLimit"]));
 	m_envInfo.setTimestamp(toInt(_o["currentTimestamp"]));
+	cnote << "ImportTest::importEnv() setLastHashes and setTimestamp ignored!";
```
Comment (CJentzsch) [inline test/TestHelper.cpp]: when you ignore them, how can you pass the tests?

File: test/TestHelper.h [hunk 6]
```diff
@@ -32,6 +32,7 @@
 #include <libevm/ExtVMFace.h>
 #include <libtestutils/Common.h>
 
+
```
Comment (CJentzsch) [inline test/TestHelper.h]: please remove empty line

File: test/TestHelper.cpp [hunk 7]
```diff
@@ -289,22 +310,22 @@ void ImportTest::exportTest(bytes const& _output, State const& _statePost)
 		m_testObject.erase(m_testObject.find("expectOut"));
 	}
 
-	// export logs
-	m_testObject["logs"] = exportLog(_statePost.pending().size() ? _statePost.log(0) : LogEntries());
+	// export logs
```
Comment (CJentzsch) [inline test/TestHelper.cpp]: there is a  space too much

File: test/TestUtils.cpp [hunk 8]
```diff
@@ -101,7 +101,9 @@ void ClientBaseFixture::enumerateClients(std::function<void(Json::Value const&,
 {
 	enumerateBlockchains([&callback](Json::Value const& _json, BlockChain const& _bc, State _state) -> void
 	{
-		FixedClient client(_bc, _state);
+		cerr << "void ClientBaseFixture::enumerateClients. FixedClient now accepts block not sate!" << endl;
```
Comment (CJentzsch) [inline test/TestUtils.cpp]: is this an error? I dont think so, I think you can remove this output.

File: test/fuzzTesting/checkRandomStateTest.cpp [hunk 9]
```diff
@@ -86,8 +87,8 @@ bool doStateTest(mValue& _v)
 		bytes output;
 
 		try
-		{
-			output = theState.execute(lastHashes(importer.m_environment.currentBlock.number()), importer.m_transaction).output;
+		{
```
Comment (CJentzsch) [inline test/fuzzTesting/checkRandomStateTest.cpp]: tabs

File: test/fuzzTesting/checkRandomStateTest.cpp [hunk 10]
```diff
@@ -139,7 +142,8 @@ bool doStateTest(mValue& _v)
 				return 1;
 			if (asserts(logs[i].data == importer.m_environment.sub.logs[i].data))
 				return 1;
-		}
+		}*/
+
```
Comment (CJentzsch) [inline test/fuzzTesting/checkRandomStateTest.cpp]: remove empty line

File: test/TestHelper.cpp [hunk 11]
```diff
@@ -105,10 +108,24 @@ ImportTest::ImportTest(json_spirit::mObject& _o, bool isFiller):
 	if (!isFiller)
 	{
 		importState(_o["post"].get_obj(), m_statePost);
-		m_environment.sub.logs = importLog(_o["logs"].get_array());
+		cnote << "ImportTest(json_spirit::mObject& _o, bool isFiller) Environment logs check ignored!";
+		//m_environment.sub.logs = importLog(_o["logs"].get_array());
 	}
 }
 
+bytes ImportTest::executeTest()
```
Comment (CJentzsch) [inline test/TestHelper.cpp]: it should not be part of the importer (who's sole purpose is to import the test json into the State objects) ot execute tests.

File: test/libethereum/state.cpp [hunk 12]
```diff
@@ -56,19 +56,18 @@ void doStateTests(json_spirit::mValue& v, bool _fillin)
...
-
-		State theState = importer.m_statePre;
+		const State importedStatePost = importer.m_statePost;
 		bytes output;
 
 		try
 		{
 			Listener::ExecTimeGuard guard{i.first};
-			output = theState.execute(lastHashes(importer.m_environment.currentBlock.number()), importer.m_transaction).output;
+			output = importer.executeTest();
```
Comment (CJentzsch) [inline test/libethereum/state.cpp]: how do the tests get the lasHashes?

**Pull request closed without merging**